src/Magnum/Shaders/Generic.h

```diff
@@ -358,7 +358,26 @@ template<UnsignedInt dimensions> struct Generic {
      */
     typedef GL::Attribute<5, Vector3> Normal;
 
-    /* 6, 7 reserved for vertex weights / bone IDs */
+    /**
+     * @brief Skin weights
+     *
+     * @ref Magnum::Vector4 "Vector4", four weights of influence per @ref JointIndices.
+     * Corresponds to @ref Trade::MeshAttribute::Weights.
+     */
+    // TODO: E.g. gltf supports importing multiple sets of 4 weights (WEIGHTS_0 [0;3], WEIGHTS_1 [4;7])
+    // I suppose these would require special shaders and it's fine to only support 4 for the GenericShaders
+    typedef GL::Attribute<6, Vector4> Weights;
+
+    /**
+     * @brief Joint indices
+     *
+     * @ref Magnum::Vector4 "Vector4", four weights of influence per @ref JointIndices.
+     * Corresponds to @ref Trade::MeshAttribute::Weights.
+     */
+    // TODO: E.g. gltf supports importing multiple sets of 4 indices (JOINTS_0 [0;3], JOINTS_1 [4;7])
+    // I suppose these would require special shaders and it's fine to only support 4 for the GenericShaders
+    // TODO: This is called "Joints" in gltf.
+    typedef GL::Attribute<7, Vector4> JointIndices;
```

Comment:
```suggestion
    typedef GL::Attribute<7, Vector4> JointIds;
```

Also isn't this an integer attribute, when it's a ID? All other attribs are singular, so maybe make this `JointId` and `Weight` also? Kinda on the fence about this tho, because there's `TextureCoordinates` also.

Reply:
Since it's always four at a time (i.e. four joints that might influence a vertex at any given time), I think conveying it's "more than one" through the name would make sence in my oppinion.